lib/acyclicGraph.ml

```diff
@@ -111,9 +113,22 @@ module Make (Point:Point) = struct
       rank : int;
       klvl: int;
       ilvl: int;
-      mutable status: status
     }
 
+  module CN = struct
+    type t = canonical_node
+    let equal x y = x == y
+    let hash x = Index.hash x.canon
+  end
+
+  module Status = Ephemeron.K1.Make(CN)
+  let status = Status.create 17
```

Comment:
We could pass a local hashtable around instead of having a global ephemeron table, then we would be thread safe.
It might even be faster since reverting statuses would just be letting the table drop out of scope.

Reply:
Indeed, that is also an option. Let's say that if this version passes CI/bench, then we try the other one. I'm quite surprised the bench did not show significant slowdown so far.

Reply:
So no, now that the code uses ephemerons for real it is super slow. I guess we can still try to pass around a map

Reply:
https://gitlab.com/coq/coq/-/jobs/2538688103